screens/modals/RewardModal.js

```diff
@@ -0,0 +1,185 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Modal, View } from 'react-native';
+import Colors from '../../assets/Colors';
+import { ButtonLabel, Title } from '../../components/BaseComponents';
+import { RewardAppliedContainer, RewardAvailableContainer } from '../../styled/checkout';
+import { ModalCenteredOpacityLayer } from '../../styled/modal';
+import { ColumnContainer, RoundedButtonContainer, RowContainer } from '../../styled/shared';
+
+export default class RewardModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      rewardsApplied: 0,
+      rewardStatus: [],
+      totalPrice: 0,
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { rewardsAvailable, rewardsApplied, totalPrice } = this.props;
+    const rewardStatus = [];
```

Comment:
Dumb that you can’t do something like : `rewardStatus = range(0, rewardsAvailable).map(...)`

You can with external libraries or unreadable one-liners but it should be in vanilla JavaScript 😠

Reply:
Okay so i chased this rabbit hole. One other way you couuuld do this is: 

```
const rewardStatus = new Array(rewardsAvailable).fill(false).map((_, i) => (...));
```

Buuuut maybe we’re getting less legible and too technical here 😂

Reply:
Okay wait so I actually tried this before (we've been iterating over a range set by some int for awhile now); forget that this status array is deprecated with the redesign for a second... 🤣 . BUT for some reason I could NOT understand, I would always get this "Array size is not a small enough positive integer" error _non-deterministically_ (dear lord) no matter how I initialized the integer, so I gave up and used a for loop ;__;